src/Tank.hpp

```diff
@@ -26,18 +29,23 @@ class TankPart {
 
 class Tank {
  public:
-  Tank(float x, float y, sf::Texture& body, sf::Texture& tower, sf::Texture& shot);
+  Tank(float x, float y, sf::Texture& body, sf::Texture& tower, sf::Texture& shot,
+       std::unique_ptr<Engine> engine);
```

Comment:
Maybe better would be to pass here unique_ptr `rvalue`? ``std::unique_ptr<Engine>&& engine`